docs/notes/router-annotations.adoc

```diff
@@ -0,0 +1,128 @@
+////
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License
+////
+
+= Router Message Annotations
+
+This document describes the implementation of the meta-data included
+in messages passed between router nodes (interior and edge).
+
+== Description
+
+Each message passed between routers includes meta-data used to
+communicate per-message state. This data is used by routers along the
+forwarding path for various reasons.  Currently this information
+includes:
+
+* The identifier for the ingress router (the router attached to the
+original sending endpoint).
```

Comment:
To be clear.  The ingress is the first _Interior_ router in the trace.  An edge cannot be an ingress router.

Reply:
Good point - doc updated.
@ted-ross since the trace-list (currently) does not include edge-mode routers I was thinking we could drop the ingress-router annotation entirely and instead use the first element in the trace-list for the ingress-router.  An optimization and reduction of duplicate information.  What do you think?

Reply:
This was considered and rejected in the past.  I don't remember why.

Reply:
I suggest that we leave this as it is.  The reason is that we may in the near future wish to include the edge-mode routers in the trace.  We can handle this by simply ignoring unrecognized trace-ids when building the trace-mask.  If we do this, the "ingress" router will not necessarily be the first router in the trace.